webapp/channels/src/components/autocomplete_selector.tsx

```diff
@@ -1,16 +1,40 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import PropTypes, {Requireable} from 'prop-types';
+import React, {ChangeEvent, PureComponent, ReactNode, RefObject} from 'react';
 
 import SuggestionBox from 'components/suggestion/suggestion_box';
 import SuggestionList from 'components/suggestion/suggestion_list';
 
-export default class AutocompleteSelector extends React.PureComponent {
+interface AutocompleteSelectorProps {
+    providers: any[];
+    value: string;
+    onSelected?: (selected: any) => void;
+    label?: ReactNode;
+    labelClassName?: string;
+    inputClassName?: string;
+    helpText?: ReactNode;
+    placeholder?: string;
+    footer?: ReactNode;
+    disabled?: boolean;
+    toggleFocus?: (focus: boolean) => void;
+    listComponent?: React.ElementType;
+    listPosition?: string;
+}
+
+interface AutocompleteSelectorState {
+    input: string;
+    focused: boolean;
+}
+
+export default class AutocompleteSelector extends PureComponent<
+AutocompleteSelectorProps,
+AutocompleteSelectorState
+> {
     static propTypes = {
-        providers: PropTypes.array.isRequired,
-        value: PropTypes.string.isRequired,
+        providers: PropTypes.array.isRequired as Requireable<any[]>,
+        value: PropTypes.string.isRequired as Requireable<string>,
```

Comment:
we need to remove proptypes when migrating to typescript. Can you check this reference PR  https://github.com/mattermost/mattermost/pull/23325